pkg/cmd/cli/cmd/debug.go

```diff
@@ -0,0 +1,565 @@
+
+	debugPodAnnotationSourceContainer = "debug.openshift.io/source-container"
+	debugPodAnnotationSourceResource  = "debug.openshift.io/source-resource"
+)
+
+// NewCmdDebug creates a command for debugging pods.
+func NewCmdDebug(fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &DebugOptions{
+		Timeout: 30 * time.Second,
+		Command: []string{"/bin/sh"},
+		Attach: kcmd.AttachOptions{
+			In:    in,
+			Out:   out,
+			Err:   errout,
+			TTY:   true,
+			Stdin: true,
+
+			Attach: &kcmd.DefaultRemoteAttach{},
+		},
+		LogsForObject: f.LogsForObject,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "debug RESOURCE/NAME [ENV1=VAL1 ...] [-c CONTAINER] [-- COMMAND]",
+		Short:   "Launch a new instance of a pod for debugging",
+		Long:    debugLong,
+		Example: fmt.Sprintf(debugExample, fmt.Sprintf("%s debug", fullName)),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(cmd, f, args, in, out, errout))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Debug())
+		},
+	}
+
+	// TODO: when T is deprecated use the printer, but keep these hidden
+	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|wide|name|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=... See golang template [http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template [http://releases.k8s.io/HEAD/docs/user-guide/jsonpath.md].")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	cmd.Flags().String("template", "", "Template string or path to template file to use when -o=go-template, -o=go-template-file. The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].")
+	cmd.Flags().Bool("no-headers", false, "When using the default output, don't print headers.")
+	cmd.Flags().MarkHidden("no-headers")
+	cmd.Flags().String("sort-by", "", "If non-empty, sort list types using this field specification.  The field specification is expressed as a JSONPath expression (e.g. 'ObjectMeta.Name'). The field in the API resource specified by this JSONPath expression must be an integer or a string.")
+	cmd.Flags().MarkHidden("sort-by")
+	cmd.Flags().Bool("show-all", true, "When printing, show all resources (default hide terminated pods.)")
+	cmd.Flags().MarkHidden("show-all")
+
+	cmd.Flags().BoolVarP(&options.NoStdin, "no-stdin", "I", options.NoStdin, "Force a pseudo-terminal to be allocated")
+	cmd.Flags().BoolVarP(&options.ForceTTY, "tty", "t", false, "Force a pseudo-terminal to be allocated")
+	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", false, "Disable pseudo-terminal allocation")
+
+	cmd.Flags().StringVarP(&options.Attach.ContainerName, "container", "c", "", "Container name; defaults to first container")
+	cmd.Flags().BoolVar(&options.KeepLabels, "keep-labels", false, "Keep the original pod labels")
```

Comment:
I think you should note that you'll usually want to pair this with keeping your readiness check.